stock_reserve_area/models/stock_reserve_area.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2023 ForgeFlow SL.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class StockReserveArea(models.Model):
+    _name = "stock.reserve.area"
+
+    name = fields.Char()
+    active = fields.Boolean(default=True)
+    company_id = fields.Many2one("res.company")
+    location_ids = fields.One2many(
+        "stock.location",
+        "reserve_area_id",
+        domain="[('reserve_area_id', '=', False),"
```

Comment:
Add to the domain the locations that belong to the company indicated in the reserve area